core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql

```diff
@@ -34,11 +35,23 @@
 
   {% if existing_relation is none %}
       {% set build_sql = get_create_table_as_sql(False, target_relation, sql) %}
+
   {% elif full_refresh_mode %}
       {% set build_sql = get_create_table_as_sql(False, intermediate_relation, sql) %}
       {% set need_swap = true %}
+
   {% else %}
-    {% do run_query(get_create_table_as_sql(True, temp_relation, sql)) %}
+    {% set build_sql %}
+      {{sql}} 
+
+      {#-- If an incremental key was provided, append the incremental filtering logic to the model sql --#}
+      {% if incremental_key is not none %}
+        -- this filter will only be applied on an incremental run
+        WHERE {{ incremental_key }} >= (SELECT MAX(THIS.{{ incremental_key }}) FROM {{ this }} THIS)
```

Comment:
You include the `WHERE` so `incremental_key` would only work for models which have no other filters (otherwise you'd end up with multiple `WHERE` clauses, which is a syntax error). Is there a way to more gracefully handle that situation, or is `incremental_key` intended only for the simplest of use cases?